src/cdk/table/table.ts

```diff
@@ -186,7 +196,10 @@ export interface RenderRow<T> {
   // declared elsewhere, they are checked when their declaration points are checked.
   // tslint:disable-next-line:validate-decorators
   changeDetection: ChangeDetectionStrategy.Default,
-  providers: [{provide: CDK_TABLE, useExisting: CdkTable}]
+  providers: [
+    {provide: CDK_TABLE, useExisting: CdkTable},
+    {provide: _VIEW_REPEATER_STRATEGY, useClass: DisposeViewRepeaterStrategy},
```

Comment:
Providing this directly might still end up with the problem of always including the code for `DisposeViewRepeaterStrategy` even when it's not being used. Is this a temporary thing until we have a directive API for switching between rending strategies?

Reply:
Yes, this is temporary. The virtual scroll table PR will introduce a directive-based API to replace it.

Reply:
Cool- can you add a TODO here to make that explicit?

Reply:
Done